bld.bat file is added for creating bwidget package in a windows conda…

nandandevinagara opened the pull request:
… environment.

<!--
Thank you for pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->

nandandevinagara committed:
bld.bat file is added for creating bwidget package in a windows conda environment.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.


I do have some suggestions for making it better though...


For **recipe**:

 * License is not an SPDX identifier (or a custom LicenseRef) nor an SPDX license expression.

Documentation on acceptable licenses can be found [here]( https://conda-forge.org/docs/maintainer/adding_pkgs.html#spdx-identifiers-and-expressions ).

nandandevinagara committed:
windows build is enabled.
test line is added for windows build.

abloemert commented:
@conda-forge-admin, please rerender

Commit:
MNT: Re-rendered with conda-build 3.21.4, conda-smithy 3.12, and conda-forge-pinning 2021.10.06.08.16.35

nandandevinagara committed:
Errors from bld.bat file is removed

setting TCL_LIB_PATH had a problem( = ) which is now rectified.
As meta.yaml file instructs the extraction of bwidget files into src folder, "src\" is prefixed in xcopy command.

nandandevinagara committed:
Build number is changed to 1 as version of bwidget is still same and recipe is updated.

nandandevinagara committed:
test command for windows is updated.

simple package require command is used.

nandandevinagara committed:
xcopy command is updated with correct dest. path

nandandevinagara committed:
Destination path of bwidget is changed

nandandevinagara committed:
TCL_LIB_PATH now contains Library in its path.

nandandevinagara committed:
test command for windows build is updated because simply importing bwidget opens a Console

fhoehle commented on recipe/bld.bat:
```diff
@@ -0,0 +1,5 @@
+set TCL_LIB_PATH=%PREFIX%\Library\lib\tcl8.6\%PKG_NAME%%PKG_VERSION%
+mkdir %TCL_LIB_PATH%
```
Please add a safety net checking if the directory exists. Or add a line raising errors: `if errorlevel 1 exit 1` see here https://github.com/conda-forge/r-base-feedstock/blob/19c643d97f43050424be0686a4d22dd88fcc5e36/recipe/bld-r-base.bat#L18

the following lines could also need this precaution.

nandandevinagara replied:
Thanks for the hint @fhoehle. I will add the error-checking commands. 
I didn't plan to switch to bwidget build in r-base.

nandandevinagara replied:
I have implemented the necessary changes. Could you please review it, @fhoehle

fhoehle replied:
Thanks adding

fhoehle reviewed (changes requested):
Thanks for adding windows builds. I would like to have some error checking statements in `bld.bat` to ensure potential errors are raised. When this PR is merged, do you plan switch to this `bwiget` build in `r-base`? See here: https://github.com/conda-forge/r-base-feedstock/blob/e74584ad023e552247fc9451a6ce2e576be2b439/recipe/meta.yaml#L136

nandandevinagara committed:
Error checking commands are added to bld.bat file

nandandevinagara committed:
Added command to copy the license .txt file into the package

nandandevinagara committed:
Source path of license.txt is updated in bld.bat
